src/main-oe/sgxlkl_params.c

```diff
@@ -38,285 +43,107 @@ struct sgxlkl_config_elem
     };
 };
 
+typedef struct json_callback_data
+{
+} json_callback_data_t;
+
+#define DEFAULT_SGXLKL_VERBOSE 0
+#define DEFAULT_SGXLKL_CWD "/"
+#define DEFAULT_SGXLKL_GW4 "10.0.1.254"
+/* The default heap size will only be used if no heap size is specified and
+ * either we are in simulation mode, or we are in HW mode and a key is provided
+ * via SGXLKL_KEY.
+ */
+#define DEFAULT_SGXLKL_OE_HEAP_PAGE_COUNT 8192 /* 8192 * 4K = 32MB */
+#define DEFAULT_SGXLKL_HEAP_SIZE 200 * 1024 * 1024
+#define DEFAULT_SGXLKL_HOSTNAME "lkl"
+#define DEFAULT_SGXLKL_IP4 "10.0.1.1"
+#define DEFAULT_SGXLKL_MASK4 "24"
+#define DEFAULT_SGXLKL_MAX_USER_THREADS 256
+#define DEFAULT_SGXLKL_ESLEEP 16000
+#define DEFAULT_SGXLKL_ETHREADS 1
+#define DEFAULT_SGXLKL_ESPINS 500
+#define DEFAULT_SGXLKL_STACK_SIZE 512 * 1024
+#define DEFAULT_SGXLKL_SWIOTLB 1
+#define DEFAULT_SGXLKL_TAP "sgxlkl_tap0"
+#define DEFAULT_SGXLKL_WG_IP "10.0.2.1"
+#define DEFAULT_SGXLKL_WG_PORT 56002
+#define DEFAULT_SGXLKL_KERNEL_CMD "mem=32M"
+#define DEFAULT_SGXLKL_HOSTNET false
+#define DEFAULT_SGXLKL_TAP_MTU 0
+
+// clang-format off
 static struct sgxlkl_config_elem sgxlkl_config[] = {
-    {"SGXLKL_APP_CONFIG",
-     "app_config",
-     TYPE_JSON,
-     {.def_char = NULL},
-     0}, /* 0 */
-
-    {"SGXLKL_CMDLINE", "cmdline", TYPE_CHAR, {.def_char = "mem=32M"}, 0},
-    /* 1 */ /* set LKL mem to 32MB (default is 64MB) */
-
-    {"SGXLKL_CWD",
-     "cwd",
-     TYPE_CHAR,
-     {.def_char = DEFAULT_SGXLKL_CWD},
-     0}, /* 2 */
-
-    {"SGXLKL_DEBUGMOUNT",
-     "debugmount",
-     TYPE_CHAR,
-     {.def_char = NULL},
-     0}, /* 3 */
-
-    {"SGXLKL_ESPINS",
-     "espins",
-     TYPE_UINT,
-     {.def_uint = {DEFAULT_SGXLKL_ESPINS, ULONG_MAX}},
-     0}, /* 4 */
-
-    {"SGXLKL_ESLEEP",
-     "esleep",
-     TYPE_UINT,
-     {.def_uint = {DEFAULT_SGXLKL_ESLEEP, ULONG_MAX}},
-     0}, /* 5 */
-
-    {"SGXLKL_ETHREADS",
-     "ethreads",
-     TYPE_UINT,
-     {.def_uint = {1, MAX_SGXLKL_ETHREADS}},
-     0}, /* 6 */
-
-    {"SGXLKL_ETHREADS_AFFINITY",
-     "ethreads_affinity",
-     TYPE_CHAR,
-     {.def_char = NULL},
-     0}, /* 7 */
-
+    {}, /* 0 */
```

Comment:
We could get rid of the empty slots.

Reply:
I was wondering why anybody ever kept them... done!